bitnami/geode/templates/locator/statefulset.yaml

```diff
@@ -0,0 +1,387 @@
+          {{- end }}
+          volumeMounts:
+            - name: data
+              mountPath: /bitnami/geode
+        {{- end }}
+        {{- if .Values.locator.initContainers }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.locator.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+      {{- end }}
+      containers:
+        - name: geode
+          image: {{ include "geode.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.locator.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.locator.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          {{- else if .Values.locator.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.locator.command "context" $) | nindent 12 }}
+          {{- else if .Values.auth.tls.enabled }}
+          command:
+            - /bin/bash
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.locator.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.locator.args "context" $) | nindent 12 }}
+          {{- else if .Values.auth.tls.enabled }}
+          args:
+            - -ec
+            - |
+              ID="${MY_POD_NAME#"{{ $fullname }}-"}"
+              mkdir -p /opt/bitnami/geode/config/certs
+              if [[ -f "/certs/geode.truststore.jks" ]] && [[ -f "/certs/geode-locator-${ID}.truststore.jks" ]]; then
+                  cp "/certs/geode.truststore.jks" "/opt/bitnami/geode/config/certs/geode-truststore.jks
+                  cp "/certs/geode-locator-${ID}.truststore.jks" "/opt/bitnami/geode/config/certs/geode.keystore.jks"
+              else
+                  echo "Couldn't find the expected Java Key Stores (JKS) files! They are mandatory when encryption via TLS is enabled."
+                  exit 1
+              fi
+              exec /opt/bitnami/scripts/geode/entrypoint.sh /opt/bitnami/scripts/geode/run.sh
+          {{- end }}
+          env:
+            - name: BITNAMI_DEBUG
+              value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
+            - name: MY_POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: GEODE_NODE_NAME
+              value: "$(MY_POD_NAME)"
+            - name: GEODE_NODE_TYPE
+              value: "locator"
+            - name: GEODE_ADVERTISED_HOSTNAME
+              value: "$(MY_POD_NAME).{{ $fullname }}-locator-hl.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}"
```

Comment:
Why `-hl` instead of `-headless`? Is it due to any DNS length limitation?

Reply:
Correct, It's just about using a shorter svc name.

Reply:
But would it cause any problem if it's longer than X characters? Just to understand this. This mainly concerns me due to it not being common amongst our chart catalog.

Reply:
Correct, indeed we're using `trunc` in the headless service manifest (check line 4)